docs/interactions.rst

```diff
@@ -0,0 +1,129 @@
+:orphan:
+
+.. currentmodule:: nextcord
+
+.. _client_index:
+   
+Interaction Commands
+======================
+
+Since Discord has added interaction commands a feature with many possibilities, we at Nextcord have decided to add them to our fleet of features.
+
+This doc will explain the innerworkings and how to use interaction commmands.
+
+We suggest you learn how to make regular commands before looking through here, we suggest looking at the :doc:`quickstart`
+
+How To Make A Simple Interaction Command
+------------------------------------------
+
+This right here is a simple ping pong command made with Nextcords slash feature.
+
+.. code-block:: python3
+
+    @bot.slash_command(name="ping")
+    async def ping(interaction):
+        await interaction.response.send_message("Pong!")
+        
+The way it works is that you use the :meth:`~Client.slash_command` function to interact with the Discord API. The ``name`` parameter is the name of your slash command.
+
+``guild_ids`` is used to limit the guilds that the slash command is available to. This is also useful for testing, as global slash commands can take up to an hour to register.
+
+Example:
+
+.. code-block:: python3
+      
+    @bot.slash_command(guild_ids=[id1, id2])
+
+How To Use Sub-Commands
+-------------------------
+
+The way sub-commands work is that you make a normal slash command that will never be called, and then make the sub-commands and have them do the work of real slash commands. There is no difference with slash commands and sub-commands. The only thing you will need to change is functions.
+
+As shown in the demonstration below you make a main slash command or a dummy slash command and build sub-commands off it.
+
+.. code-block:: python3
+
+    @bot.slash_command()
+    async def main(interaction):
+        await interaction.response.send_message("This will never get called if this has subcommands.")
+
+
+    @main.subcommand()
+    async def sub1(interaction):
+        await interaction.response.send_message("This is subcommand 1!")
+
+
+    @main.subcommand(name="sub2", description="Sub2s Description")
+    async def subcommand_two(
+                            interaction: Interaction,
```

Comment:
maybe only indent these args by 4 spaces instead as the ones that line with opening look like 
```py
def aaa(arg
        arg2):
```

and with no first arg it should be
```py
def aaa(
    arg,
    arg2
):
```
or 
```py
def aaa(
    arg, arg2
):
```

Reply:
like you did on L73